workflow-examples/src/test/java/com/redhat/parodos/examples/integration/KubeapiWorkFlow.java

```diff
@@ -0,0 +1,130 @@
+
+			// EXECUTE WORKFLOW
+			WorkflowApi workflowApi = new WorkflowApi();
+
+			// 1 - Define WorkRequests
+			WorkRequestDTO workGet = WorkRequestDTO.builder().workName(taskName)
+					.arguments(Arrays.asList(new ArgumentRequestDTO().key("api-group").value(""),
+							new ArgumentRequestDTO().key("api-version").value("v1"),
+							new ArgumentRequestDTO().key("kind-plural-name").value("configmaps"),
+							new ArgumentRequestDTO().key("operation").value("get"),
+							new ArgumentRequestDTO().key("resource-name").value("my-cm"),
+							new ArgumentRequestDTO().key("resource-namespace").value("test"),
+							new ArgumentRequestDTO().key("work-ctx-key").value("theKey")))
+					.build();
+
+			WorkRequestDTO workCreate = WorkRequestDTO.builder().workName(taskName).arguments(Arrays.asList(
+					new ArgumentRequestDTO().key("api-group").value(""),
+					new ArgumentRequestDTO().key("api-version").value("v1"),
+					new ArgumentRequestDTO().key("kind-plural-name").value("configmaps"),
+					new ArgumentRequestDTO().key("operation").value("create"),
+					new ArgumentRequestDTO().key("resource-json").value("{"
+							+ "\"apiVersion\": \"v1\",\"data\": {\"a\": \"vala\",\"b\": \"valb\"},\"kind\": \"ConfigMap\","
+							+ "\"metadata\": {\"name\": \"my-cm-create\",\"namespace\": \"test\"}}")))
+					.build();
+
+			WorkRequestDTO workUpdate = WorkRequestDTO.builder().workName(taskName).arguments(Arrays.asList(
+					new ArgumentRequestDTO().key("api-group").value(""),
+					new ArgumentRequestDTO().key("api-version").value("v1"),
+					new ArgumentRequestDTO().key("kind-plural-name").value("configmaps"),
+					new ArgumentRequestDTO().key("operation").value("update"),
+					new ArgumentRequestDTO().key("resource-json").value("{"
+							+ "\"apiVersion\": \"v1\",\"data\": {\"a\": \"valb\",\"b\": \"vala\"},\"kind\": \"ConfigMap\","
+							+ "\"metadata\": {\"name\": \"my-cm-create\",\"namespace\": \"test\"}}")))
+					.build();
+
+			// 2 - Define WorkFlowRequests
+			WorkFlowRequestDTO workFlowRequestGet = WorkFlowRequestDTO.builder().projectId(testProject.getId())
+					.workFlowName(workflowName).works(Arrays.asList(workGet)).build();
+			WorkFlowRequestDTO workFlowRequestCreate = WorkFlowRequestDTO.builder().projectId(testProject.getId())
+					.workFlowName(workflowName).works(Arrays.asList(workCreate)).build();
+			WorkFlowRequestDTO workFlowRequestUpdate = WorkFlowRequestDTO.builder().projectId(testProject.getId())
+					.workFlowName(workflowName).works(Arrays.asList(workUpdate)).build();
+
+			WorkFlowRequestDTO workFlowRequests[] = new WorkFlowRequestDTO[] { workFlowRequestGet,
+					workFlowRequestCreate, workFlowRequestUpdate };
+
+			// 3 - Execute WorkFlowRequests
+			for (WorkFlowRequestDTO workFlowRequest : workFlowRequests) {
+				WorkFlowResponseDTO execute = workflowApi.execute(workFlowRequest);
+
+				assertNotNull(execute.getWorkFlowExecutionId());
```

Comment:
should the result be checked as well (the status) or the assumption is that in case of a failure, an exception will be caught and fail the test?

Reply:
and again it's just an example. 
Each meaningful example class should test all the scenarios to ensure a proper execution.

Reply:
In case of status 'FAILED' the execution ID will be null. So there's no point asserting them both